torch/csrc/profiler/util.cpp

```diff
@@ -115,20 +106,47 @@ std::string stacksToStr(
   return "\"" + rc + "\"";
 }
 
+std::vector<std::vector<int64_t>> flattenList(c10::List<c10::IValue> list, std::string fn_name) {
+  std::vector<std::vector<int64_t>> tensor_dims;
+  int idx = 0;
+  for (const c10::IValue input : list) {
+    if (input.isTensor()) {
+      const at::Tensor& tensor = input.toTensor();
+      if (tensor.defined()) {
+        tensor_dims.push_back(input.toTensor().sizes().vec());
+      }
+    }
+    idx += 1;
+  }
+  return tensor_dims;
+}
+
 std::vector<std::vector<int64_t>> inputSizes(const at::RecordFunction& fn) {
   std::vector<std::vector<int64_t>> sizes;
   sizes.reserve(fn.inputs().size());
+  int iter = 0;
   for (const c10::IValue& input : fn.inputs()) {
     if (!input.isTensor()) {
-      sizes.emplace_back();
-      continue;
-    }
-    const at::Tensor& tensor = input.toTensor();
-    if (tensor.defined()) {
-      sizes.push_back(input.toTensor().sizes().vec());
+      if (input.isList()) {
```

Comment:
nit: At this point is should probably be `if (input.isTensor) { ... } else if (input.isList()) { ... } else {}`
Also, can we add a `flatten_lists` arg so we don't change the behavior of other profilers? Otherwise we could break users who previously ignored lists, and now have no way to reconstruct the old argument setup.

Reply:
I simplified the if clause like you proposed.
Also I added a flag to inputSizes  `flatten_list_enabled`.   It is false by default.  It is set to true by nvtx_observer.  When using nvtx_observer users can't directly query the profiler, so there shouldn't be existing code that is testing the shape string.  In the timeline view I need to see the flattened list in order to know the tensor shapes that were passed to the concat op.